projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -283,6 +293,16 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
     this.setupEventListeners();
   }
 
+  private redrawVisualization(): void {
+    const chartViz = select(this.chartContainerElement!).selectAll(
```

Comment:
In d3, usually you don't remove nodes in order to redraw them - you update the bound data and let d3 optimize the dom manipulation.

Reply:
I looked into the code and tried to modify the code to use d3 update pattern but it seems that, we're using some underlying classes like band, axis, legend which are not using the same. To update this behaviour we need to update at many places, not sure about that much of effort!

Reply:
Those classes shouldn't be either, but also should be independent of what we're doing with legend. The main thing I'd go for here is to remove the special casing for legend with the boolean - that's the bigger code smell, the update vs remove/redraw pattern is easy to fix later as long as it is contained within drawLegend.

That boolean, from my recollection was needed because we don't want to rebuild the legend now that it's holding state. It would be better to address that and not rebuild it but moving this check inside drawLegend so it's self contained works too if we need a quicker way.

Reply:
What I have done here is, I broke the updateData function and apart from legend part, I put in another function called `drawVisualizations()`, So I am calling that function from `redrawVisualization`, by which we're not doing this boolean part. Now the part where not removing the chart Visualization and using D3 update flow, we can skip as mentioned above as of now.